### PR TITLE
fix(monitors): Adds unique_together for MonitorEnvironment

### PR DESCRIPTION
**Pull request by rjo100**
Dependent on:
https://github.com/getsentry/getsentry/pull/9731

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
Migration looks good to me. Let's make sure there's no duplicates left once we the required cleanup is run

File: src/sentry/models/monitorenvironment.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_monitorenvironment"
         indexes = [models.Index(fields=["monitor", "environment"])]
```
Comment (wedamija) [inline src/sentry/models/monitorenvironment.py:31]: We could also remove this index since the unique will do the same thing

**Review by wedamija — APPROVED** [review]
lgtm, this table has only 2k rows so it's fine to add this index during deploy.

File: src/sentry/models/monitorenvironment.py (hunk 1 above)
Comment (rjo100) [reply src/sentry/models/monitorenvironment.py:31]: ah...smart

**Comment by rjo100** [comment]
Confirmed no duplicates
https://redash.getsentry.net/queries/3745/source

**Pull request merged**